src/main/webapp/app/exercises/quiz/manage/drag-and-drop-question/drag-and-drop-question-edit.component.ts

```diff
@@ -875,4 +887,112 @@ export class DragAndDropQuestionEditComponent implements OnInit, OnChanges, Afte
         this.cleanupQuestion();
         this.markdownEditor.parse();
     }
+
+    /**
+     * Create new drag items for each drop location in the background image
+     * @returns
+     */
+    getImagesFromDropLocations() {
+        for (const someLocation of this.question.dropLocations!) {
+            const image = new Image();
+            let dataUrl: string = '';
+            let bgWidth;
+            let bgHeight;
+            image.onload = () => {
+                bgHeight = image.height;
+                bgWidth = image.width;
+
+                const canvas = document.createElement('canvas');
+                const context = canvas.getContext('2d');
+
+                if (context) {
+                    const scalarHeight = bgHeight / 200;
+                    const scalarWidth = bgWidth / 200;
+                    canvas.width = someLocation.width! * scalarWidth;
+                    canvas.height = someLocation.height! * scalarHeight;
+                    context.drawImage(
+                        image,
+                        someLocation.posX! * scalarWidth,
+                        someLocation.posY! * scalarHeight,
+                        someLocation.width! * scalarWidth,
+                        someLocation.height! * scalarHeight,
+                        0,
+                        0,
+                        someLocation.width! * scalarWidth,
+                        someLocation.height! * scalarHeight,
+                    );
+
+                    dataUrl = canvas.toDataURL('image/png');
+                    const dragItemCreated = this.createImageDragItemFromFile(this.dataUrlToFile(dataUrl, 'placeholder' + someLocation.posX!))!;
+                    const dndMapping = new DragAndDropMapping(dragItemCreated, someLocation);
+                    this.question.correctMappings!.push(dndMapping);
+                }
+            };
+            image.src = this.backgroundImage.src;
+        }
+        this.blankOutBackgroundImage();
+    }
+
+    /**
+     * Takes all drop locations and replaces their location with a white rectangle on the background image
+     */
+    blankOutBackgroundImage() {
+        const backGroundBlankingCanvas = document.createElement('canvas');
+        const backGroundBlankingContext = backGroundBlankingCanvas.getContext('2d');
```

Comment:
To keep it consistent, I'd keep `background` in all lower case here. (like in the method name)